From bcb8b2feee2ddc32f15979539d8ecab5017ba980 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 10 Feb 2015 08:27:42 +0800
Subject: [PATCH] MLK-10238-3: ARM: imx: gpc: correct the function and
 structure

Correct the function and structure since update the below patch
(24d70aa Revert "base: power: Add generic OF-based power domain look-up")
(dc092bc PM / Domains: Add generic OF-based PM domain look-up)
Signed-off-by: Robin Gong <b38343@freescale.com>
---
 arch/arm/mach-imx/gpc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-imx/gpc.c b/arch/arm/mach-imx/gpc.c
index 0f31e86..2d13eb1 100644
--- a/arch/arm/mach-imx/gpc.c
+++ b/arch/arm/mach-imx/gpc.c
@@ -488,7 +488,7 @@ static struct generic_pm_domain *imx_gpc_domains[] = {
 
 static struct genpd_onecell_data imx_gpc_onecell_data = {
 	.domains = imx_gpc_domains,
-	.domain_num = ARRAY_SIZE(imx_gpc_domains),
+	.num_domains = ARRAY_SIZE(imx_gpc_domains),
 };
 
 static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
@@ -540,7 +540,7 @@ static int imx_gpc_genpd_init(struct device *dev, struct regulator *pu_reg)
 	pm_genpd_init(&imx6q_pu_domain.base, NULL, is_off);
 	pm_genpd_init(&imx6s_display_domain.base, NULL, is_off);
 
-	return of_genpd_add_provider(dev->of_node, of_genpd_xlate_onecell,
+	return __of_genpd_add_provider(dev->of_node, __of_genpd_xlate_onecell,
 				     &imx_gpc_onecell_data);
 
 }
-- 
1.8.0

